api/src/main/java/jakarta/data/repository/Pageable.java

```diff
@@ -129,14 +145,59 @@ public static Pageable page(long page) {
      * @param page The page
      * @param size The size
      * @return The pageable
+     * @throws IllegalArgumentException when page or size are negative
      */
     public static Pageable of(long page, long size) {
+        return of(page, size, Collections.emptyList());
+    }
+
+    /**
+     * Creates a new Pageable at the given size, page and Sort
+     *
+     * @param page The page
+     * @param size The size
+     * @param sort the sort
+     * @return The pageable
+     * @throws IllegalArgumentException when page or size are negative
+     */
+    public static Pageable of(long page, long size, Sort sort) {
```

Comment:
Let's not tightly couple dynamic Sort with pagination.  It is best left decoupled as it was prior to these changes, where it is possible for the user to do both of:

```
page = products.findByNameLikeOrderByPriceDesc(pattern, pageable);
page = products.findByNameLike(pattern, pageable, Sort.desc("price"));
```

Reply:
I don't think that makes sense. Of course, `Sort` should be usable standalone, but as soon as you use a `Pageable`, it strictly *needs* a `Sort`, as otherwise the pagination order is not stable. Being able to provide a `Sort` instance in addition to a `Pageable` this only creates confusion:

- Callers wonder about two different ways to hand in the `Sort`? Via the `Pageable`? As additional parameter? Both?
- Implementors now have to decide how to interpret the additional `Sort`? Does that one need to get integrated into the `Pageable` or has the caller already done that?

That's why we decided that methods taking a `Pageable` would never take an additional `Sort`. The caller needs to prepare the `Pageable` instance completely, including sorting directives. The called code can work solely with the `Sort` instance contained in the `Pageable`.

Reply:
> * Callers wonder about two different ways to hand in the `Sort`? Via the `Pageable`? As additional parameter? Both?

There is nothing for callers to get confused about if it isn't possible to specify `Sort` via the `Pageable`.  Then the API itself doesn't allow you to attempt an erroneous combination of `Sort` via `Pageable` and `Sort` as method parameters.  This seems like a good reason for leaving it as is and not adding `Sort` to `Pageable`.

> * Implementors now have to decide how to interpret the additional `Sort`? Does that one need to get integrated into the `Pageable` or has the caller already done that?

`Pageable` is provided by the Jakarta Data API, not the implementation.  If the API doesn't allow associating a `Sort` with a `Pageable` then there isn't any decision or confusion for implementations on the matter.  So this also seems like a reason for leaving `Pageable` as is and not adding a `Sort` to it.

> Being able to provide a `Sort` instance in addition to a `Pageable` this only creates confusion

There is only confusion if `Pageable` also has its own `Sort`, which is what I'm aiming to avoid.

Reply:
A `Pageable` without a `Sort` doesn't make any sense for the reason described above. Pagination requires stable sorting.

Reply:
@njr-11 about ```Pageable```without ```Sort```is like selling a car without an engine.


+1 for keep ```Pageable``` **with** ```Sort```

Reply:
> @njr-11 about `Pageable`without `Sort`is like selling a car without an engine.

A better analogy would be selling a car without a built-in navigation system or map and someone to drive it. Many drivers can supply their own GPS or have other ways of knowing how to get around. The concept of a car still makes sense without these things sold and provided together.
The same is true of `Pageable`. It's the information for creating a page from a large sorted dataset. You can apply the page sizes and indices to different ways of sorting, with some provided by a `Sort` and others provided by a repository method's `OrderBy...`. There are different ways to tell it how to navigate which can be provided separately.

It's true that the pagination information isn't useful without sorting/ordering, but pagination information also isn't useful without a dataset, and everyone seems to agree on not coupling the dataset to `Pageable`. It should be a reasonable possibility for design that `Sort` not be coupled to `Pageable` either.

We should at least acknowledge that both proposals do make sense and this is a matter of preference for one style of design over another. This sort of thing ought to be a vote, but observing that existing implementations have already chosen the Sort-attached-to-Pageable approach, I can see that my preference for decoupling the two won't come out ahead, so I'll go ahead and review the portions of the pull that cover adding Sort to Pageable.

Reply:
It is a good point @njr-11

Reply:
I am in agreement with @odrotbohm as stated paging does require stable sorting so I don't see value in decoupling